### PR TITLE
Apps 11646 update cld sdk

### PR DESCRIPTION
**Pull request by e1adn**
https://cloudinary.atlassian.net/browse/APPS-11646

Confluence :
https://cloudinary.atlassian.net/wiki/spaces/RDKB/pages/2071953466/Video+Player+-+using+the+new+SDK

### CODE REVIEW

File: docs/360.html [hunk 1]
```diff
@@ -29,11 +29,7 @@
 
       player = cld.videoPlayer('player', {
         publicId: 'video-player/Golden-Gate-Bridge',
-        sourceTypes: ['mp4'],
-        transformation: {
```
Comment (e1adn) [inline docs/360.html:33]: Looks like a bug. A transformation was provided but it's not as part of the URL.
it has been fixed. 
Screen Shot 2022-02-17 at 17 02 24

File: test/unit/videoSource.test.js [hunk 2]
```diff
@@ -430,7 +424,7 @@ describe('test isCodecAlreadyExist method', () => {
       });
 
       const srcs = source.generateSources();
-      expect(srcs[0].src).toEqual('http://res.cloudinary.com/demo/video/upload/vc_vp9/sea_turtle.webm');
+      expect(srcs[0].src).toEqual('http://res.cloudinary.com/demo/video/upload/w_400/vc_vp9/sea_turtle.webm');
```
Comment (e1adn) [inline test/unit/videoSource.test.js:427]: Again, looks like a bug.A transformation as provided but have not been added to the url

File: docs/360.html [hunk 3]
```diff
@@ -23,17 +21,12 @@
   </script>
 
   <script type="text/javascript">
-    window.addEventListener('load', function(){
-
-      var cld = cloudinary.Cloudinary.new({ cloud_name: 'demo' });
+    window.addEventListener('load', function() {
 
-      player = cld.videoPlayer('player', {
```
Comment (tsi) [inline docs/360.html:30]: @e1adn edited the script but not the visible code example, see here below (line ~70) and in all other example pages.

File: src/index.js [hunk 4]
```diff
@@ -1,18 +1,31 @@
 import 'assets/styles/main.scss';
-import cloudinary from 'cloudinary-core';
 import VideoPlayer from './video-player';
 import { assign } from 'utils/assign';
 
-cloudinary.VideoPlayer = VideoPlayer;
 
-const proto = cloudinary.Cloudinary.prototype;
+function videoPlayer(id, playerOptions, ready = null, cloudinaryConfig) {
```
Comment (tsi) [inline src/index.js:6]: What is the use case for the 3rd & 4th arguments?
Do we have an example using them?

File: src/plugins/cloudinary/common.js [hunk 5]
```diff
@@ -43,28 +48,16 @@ const isSrcEqual = (source1, source2) => {
   return src1 === src2;
 };
 
-const mergeCloudinaryConfig = (config1, config2) => {
-  if (config1.constructor.name === 'Cloudinary' && config1.config) {
-    config1 = config1.config();
-  }
-
-  const newConfig = new cloudinary.Cloudinary(config1);
+export const extendCloudinaryConfig = (currentConfig, newConfig) => Object.assign(currentConfig, newConfig);
```
Comment (tsi) [inline src/plugins/cloudinary/common.js:51]: Isn't this redundant? we already have an `assign` utility function.

File: src/plugins/cloudinary/index.js [hunk 6]
```diff
@@ -127,7 +132,7 @@ class CloudinaryContext extends mixin(Playlistable) {
         return _transformation;
       }
 
-      _transformation = getCloudinaryInstanceOf(cloudinary.Transformation, trans);
+      _transformation = trans;
 
       return _chainTarget;
```
Comment (tsi) [inline src/plugins/cloudinary/index.js:137]: I never understood/liked this pattern, WDYT?

File: src/utils/cloudinary.js [hunk 7]
```diff
@@ -111,17 +110,6 @@ function parseUri(str) {
   return uri;
 }
 
-function getCloudinaryInstanceOf(Klass, obj) {
```
Comment (tsi) [inline src/utils/cloudinary.js:114]: Great move

**Review by tsi — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -44,12 +41,6 @@ The Cloudinary video player offers standard and light package variations, availa
 - `cld-video-player.light.js` - Non minified version which does not include any optional modules.
 - `cld-video-player.light.min.js` - Minified version which does not include any optional modules. (for smaller bundle size)
 
-### Cloudinary JavaScript library
-
-The Core Cloudinary JavaScript library provides several classes, defined under the "`cloudinary`" domain. The reference documentation is located at https://cloudinary.github.io/pkg-cloudinary-core
-
-## Getting started
```
[the remark this answers is not in the document]
Comment (e1adn) [reply README.md:51]: ops, fixex

File: docs/360.html (hunk 3 above)
Comment (e1adn) [reply docs/360.html:30]: done

File: src/index.js (hunk 4 above)
Comment (e1adn) [reply src/index.js]: the 4th argument was for internal use. I was rewrite it

File: src/index.js [hunk 9]
```diff
@@ -2,28 +2,28 @@ import 'assets/styles/main.scss';
 import VideoPlayer from './video-player';
 import { assign } from 'utils/assign';
 
-
-function videoPlayer(id, playerOptions, ready = null) {
-  return new VideoPlayer(id, playerOptions, ready);
+if (window.cloudinary && window.cloudinary.Cloudinary) {
+  console.error('For version 2.0.0 and above, cloudinary-core is not needed for using the Cloudinary Video Player');
```
Comment (tsi) [inline src/index.js:6]: I think a `console.warning` is enough here

**Review by tsi — APPROVED** [review]

File: src/index.js [hunk 10]
```diff
@@ -1,18 +1,31 @@
...
+});
 
-proto.videoPlayers = function(selector, options = {}, ready = null) {
-  assign(options, { cloudinaryConfig: this });
-  return VideoPlayer.all(selector, options, ready);
+export const videoPlayer = getVideoPlayer();
+export const videoPlayers = getVideoPlayers();
+
+window.cloudinary = {
+  ...(window.cloudinary || {}),
```
Comment (maoznir) [inline src/index.js:24]: Make sure the spread operator is transpiled for ie11 support

**Review by maoznir — APPROVED** [review]

File: src/index.js (hunk 9 above)
Comment (maoznir) [reply src/index.js:6]: indeed

File: src/index.js (hunk 10 above)
Comment (e1adn) [reply src/index.js:24]: The code is getting compiled.

**Pull request merged**